include/kernels/color_spinor_pack.cuh

```diff
@@ -173,26 +173,26 @@ namespace quda {
     parity_dim_dir %= arg.nParity2dim_threads;
 
     const int dim_dir = parity_dim_dir % (2*dim_threads);
-    const int dim0 = dim_dir / 2;
+    const int dim0 = dim_dir / 2; // this is the dim offset for each thread =0 for dim_threads=1)
     const int dir = dim_dir % 2;
     const int parity = (arg.nParity == 2) ? (parity_dim_dir / (2*dim_threads) ) : arg.parity;
     const int spinor_parity = (arg.nParity == 2) ? parity : 0;
     const int spin_block = (spin_color_block / (Nc / Mc)) * Ms;
     const int color_block = (spin_color_block % (Nc / Mc)) * Mc;
 
 #pragma unroll
-    for (int dim=dim0; dim<4; dim+=dim_threads) {
+    for (int dim=0; dim<4; dim+=dim_threads) {
```

Comment:
Was this a bug before? Or one of those "right for the wrong reasons" fixes?

Reply:
Not a correctness bug before, but was a performance bug.  Compiler could not unroll this loop: it was only that clang-cuda actually reported the lack of loop unrolling where as nvcc was silent on the matter.  As a result we may see small performance uplift for this routine.